File: frontend/src/types/job.ts
export interface Job {
  id: string
  external_id: string
  source: string
  title: string
  company: string
  location: string | null
  description: string | null
  url: string
  posted_date: string | null
  scraped_at: string
  match_score: number | null
  score_reasoning: string | null
  cover_letter: string | null
  is_remote: boolean
  salary_range: string | null
}

export interface JobListResponse {
  items: Job[]
  total: number
  page: number
  limit: number
}

File: backend/app/schemas/job.py
import uuid
from datetime import date, datetime

from pydantic import BaseModel


class JobResponse(BaseModel):
    id: uuid.UUID
    external_id: str
    source: str
    title: str
    company: str
    location: str | None
    description: str | None
    url: str
    posted_date: date | None
    scraped_at: datetime
    match_score: int | None
    score_reasoning: str | None
    cover_letter: str | None
    is_remote: bool
    salary_range: str | None

    model_config = {"from_attributes": True}


class JobListResponse(BaseModel):
    items: list[JobResponse]
    total: int
    page: int
    limit: int


class CoverLetterUpdate(BaseModel):
    cover_letter: str
